boring/src/ssl/mod.rs

```diff
@@ -3800,11 +3441,24 @@ impl SslRef {
         Ok(())
     }
 
-    /// Sets the private key.
-    ///
-    /// This corresponds to [`SSL_use_PrivateKey`].
+    /// Sets the list of CA names sent to the client.
     ///
-    /// [`SSL_use_PrivateKey`]: https://www.openssl.org/docs/man1.1.1/man3/SSL_use_PrivateKey.html
+    /// The CA certificates must still be added to the trust root - they are not automatically set
+    /// as trusted by this method.
+    #[corresponds(SSL_set_client_CA_list)]
+    pub fn set_client_ca_list(&mut self, list: Stack<X509Name>) {
+        #[cfg(feature = "rpk")]
+        assert!(
+            !self.ssl_context().is_rpk(),
+            "This API is not supported for RPK"
+        );
+
+        unsafe { ffi::SSL_set_client_CA_list(self.as_ptr(), list.as_ptr()) }
+        mem::forget(list);
```

Comment:
Why is this legitimate?

Reply:
`SSL_set_client_CA_list` takes ownership of the list: https://github.com/google/boringssl/blob/master/include/openssl/ssl.h#L2966

Constructing the `Stack<X509Name>` requires the application to own the `Stack`, and the owned type will free the underlying memory if `drop` is called. The general pattern here is to pass ownership to boringssl, and call `mem::forget` to release that ownership without calling `drop` and destroying the underlying stack